data_ingestion/common/di_url_file_to_gcs.py

```diff
@@ -18,28 +18,40 @@ def url_file_to_gcs(url, url_params, gcs_bucket, dest_filename):
     url_params: URL parameters to be passed to requests.get().
     gcs_bucket: Name of the GCS bucket to upload to (without gs://).
     dest_filename: What to name the downloaded file in GCS. Include the file extension."""
-  try_urls_to_download_file_to_gcs([url], url_params, gcs_bucket, dest_filename)
+  download_first_url_to_gcs([url], url_params, gcs_bucket, dest_filename)
 
-def try_urls_to_download_file_to_gcs(url_list, url_params, gcs_bucket, dest_filename):
-  """Attempts to download a file from a list of potential URLs and uploads as a blob to the given GCS bucket.
+def download_first_url_to_gcs(url_list, url_params, gcs_bucket, dest_filename):
+  """Iterates over the list of potential URLs that may point to the data source until one of the URLs succeeds in downloading.
+    If no URL suceeds, the method will return an error.
 
-     url_list: List of URLs where the file may be found.
-     url_params: URL parameters to be passed to requests.get().
-     gcs_bucket: Name of the GCS bucket to upload to (without gs://).
-     dest_filename: What to name the downloaded file in GCS. Include the file extension."""
+    url_list: List of URLs where the file may be found.
+    url_params: URL parameters to be passed to requests.get().
+    gcs_bucket: Name of the GCS bucket to upload to (without gs://).
+    dest_filename: What to name the downloaded file in GCS. Include the file extension."""
+
+  # Establish connection to valid GCS bucket
+  try: 
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(gcs_bucket)
+  except google.cloud.exceptions.NotFound:
+    logging.error("GCS Bucket %s not found", gcs_bucket)
```

Comment:
Add a return after this so it doesn't try to download a file that can't be uploaded, or move to the end along with the rest of the blob code

Reply:
oops good catch